controlplane/test/contracts.test.ts

```diff
@@ -0,0 +1,764 @@
+      graphUrl: 'http://localhost:4000',
+      routingUrl: 'http://localhost:3002',
+    });
+    expect(createResp.response?.code).toBe(EnumStatusCode.OK);
+
+    await client.createContract({
+      name: contractGraphName,
+      namespace: 'default',
+      sourceGraphName: monographName,
+      excludeTags: ['test'],
+      routingUrl: 'http://localhost:8081',
+      admissionWebhookUrl: 'http://localhost:8085',
+      readme: 'test',
+    });
+
+    const moveRes = await client.moveMonograph({
+      name: monographName,
+      namespace: 'default',
+      newNamespace: prod
+    });
+    expect(moveRes.response?.code).toEqual(EnumStatusCode.OK)
+
+    const getContractRes = await client.getFederatedGraphByName({
+      name: contractGraphName,
+      namespace: prod,
+    });
+    expect(getContractRes.response?.code).toEqual(EnumStatusCode.OK)
+    expect(getContractRes.subgraphs.length).toEqual(1)
+
+    await server.close();
+  });
+
+  test('Contract is deleted on deleting source monograph', async (testContext) => {
+    const { client, server } = await SetupTest({ dbname });
+
+    const monographName = genID('monograph');
+    const contractGraphName = genID('contract');
+
+    const createResp = await client.createMonograph({
+      name: monographName,
+      namespace: 'default',
+      graphUrl: 'http://localhost:4000',
+      routingUrl: 'http://localhost:3002',
+    });
+    expect(createResp.response?.code).toBe(EnumStatusCode.OK);
+
+    await client.createContract({
+      name: contractGraphName,
+      namespace: 'default',
+      sourceGraphName: monographName,
+      excludeTags: ['test'],
+      routingUrl: 'http://localhost:8081',
+      admissionWebhookUrl: 'http://localhost:8085',
+      readme: 'test',
+    });
+
+    const deleteRes = await client.deleteMonograph({
+      name: monographName,
+      namespace: 'default',
+    });
+    expect(deleteRes.response?.code).toEqual(EnumStatusCode.OK)
+
+    const getContractRes = await client.getFederatedGraphByName({
+      name: contractGraphName,
+      namespace: 'default',
+    });
+    expect(getContractRes.response?.code).toEqual(EnumStatusCode.ERR_NOT_FOUND)
+
+    await server.close();
+  });
+
+  test('Contract is migrated on migrating monograph', async (testContext) => {
+    const { client, server } = await SetupTest({ dbname });
+
+    const monographName = genID('monograph');
+    const contractGraphName = genID('contract');
+
+    const createResp = await client.createMonograph({
+      name: monographName,
+      namespace: 'default',
+      graphUrl: 'http://localhost:4000',
+      routingUrl: 'http://localhost:3002',
+    });
+    expect(createResp.response?.code).toBe(EnumStatusCode.OK);
+
+    await client.createContract({
+      name: contractGraphName,
+      namespace: 'default',
+      sourceGraphName: monographName,
+      excludeTags: ['test'],
+      routingUrl: 'http://localhost:8081',
+      admissionWebhookUrl: 'http://localhost:8085',
```

Comment:
Did you test that admission config is optional?

Reply:
It is optional. Some tests I do not pass it